sdk/dotnet/Pulumi.Automation/Runtime/LanguageRuntimeService.cs

```diff
@@ -23,10 +24,69 @@ public LanguageRuntimeService(CallerContext callerContext)
             this._callerContext = callerContext;
         }
 
-        public override Task<GetRequiredPluginsResponse> GetRequiredPlugins(GetRequiredPluginsRequest request, ServerCallContext context)
+        public override async Task<GetRequiredPluginsResponse> GetRequiredPlugins(GetRequiredPluginsRequest request, ServerCallContext context)
         {
             var response = new GetRequiredPluginsResponse();
-            return Task.FromResult(response);
+            var assembly = this._callerContext.Program.DiscoveryAssembly;
+
+            // if discovery assembly is null, assuming consumer installed plugins manually
+            if (assembly is null)
+                return response;
+
+            var pulumiAssemblyNames = assembly.GetReferencedAssemblies().Where(x => x.Name != null && x.Name.StartsWith("Pulumi.")).ToArray();
```

Comment:
`StartsWith` does a culture-sensitive comparison by default. If we do keep this implementation, we should use an ordinal comparison here:

```suggestion
            var pulumiAssemblyNames = assembly.GetReferencedAssemblies().Where(x => x.Name != null && x.Name.StartsWith("Pulumi.", StringComparison.Ordinal)).ToArray();
```